proto/frequenz/api/dispatch/dispatch.proto

```diff
@@ -90,7 +90,7 @@ message DispatchFilter {
   repeated DispatchComponentSelector selectors = 4;
 
   // Filter by time interval
-  TimeIntervalFilter time_interval = 5;
+  optional TimeIntervalFilter time_interval = 5;
```

Comment:
Marking non-primitive data-members as `optional` is redundant. Doing this leads readers to other rabbit holes, like if something is not marked `optional`, should it be marked as `required`? How do I read a non-primitive type that is neither `optional` nor `required`? Since non-primitive types are optional by default in proto3, I think we should not do this :)